### PR TITLE
Verifying Singleton Paymaster Signature Replay Attack

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: test/paymaster/singleton/verifying-singleton-paymaster.ts [hunk 1]
```diff
@@ -117,7 +117,11 @@ describe("EntryPoint with VerifyingPaymaster", function () {
       entryPoint
     );
 
-    const hash = await verifyingSingletonPaymaster.getHash(userOp1);
+    const hash = await verifyingSingletonPaymaster.getHash(
+      userOp1,
+      0,
```
Comment (filmakarov) [inline test/paymaster/singleton/verifying-singleton-paymaster.ts:122]: I would suggest actually getting nonce from the contract here

same on line 160-164

File: contracts/smart-contract-wallet/paymasters/verifying/singleton/VerifyingSingletonPaymaster.sol [hunk 2]
```diff
@@ -142,8 +146,6 @@ contract VerifyingSingletonPaymaster is BasePaymaster {
         paymasterNonces[userOp.getSender()]++;
```
Comment (filmakarov) [inline contracts/smart-contract-wallet/paymasters/verifying/singleton/VerifyingSingletonPaymaster.sol:146]: probably change to` ++paymasterNonces[userOp.getSender()]` according to gas optimizations reports

File: test/paymaster/singleton/verifying-singleton-paymaster.ts (hunk 1 above)
Comment (livingrockrises) [reply test/paymaster/singleton/verifying-singleton-paymaster.ts:122]: done

File: contracts/smart-contract-wallet/paymasters/verifying/singleton/VerifyingSingletonPaymaster.sol (hunk 2 above)
Comment (livingrockrises) [reply contracts/smart-contract-wallet/paymasters/verifying/singleton/VerifyingSingletonPaymaster.sol]: done

File: test/paymaster/singleton/verifying-singleton-paymaster.ts (hunk 1 above)
Comment (livingrockrises) [reply test/paymaster/singleton/verifying-singleton-paymaster.ts]: if looks good do approve

**Review by AmanRaj1608 — APPROVED** [review]

**Pull request merged**